expr/expression.go

```diff
@@ -47,12 +53,18 @@ func ExprFromProto(e *proto.Expression, baseSchema types.Type, ext ExtensionLook
 			}
 		}
 
-		id, ok := ext.DecodeFunc(et.ScalarFunction.FunctionReference)
-		if !ok {
-			return nil, substraitgo.ErrNotFound
-		}
+		var (
+			id   extensions.ID
+			decl *extensions.ScalarFunctionVariant
+		)
+		if ext != nil {
+			var ok bool
+			if id, ok = ext.DecodeFunc(et.ScalarFunction.FunctionReference); !ok {
+				return nil, substraitgo.ErrNotFound
+			}
 
-		decl, _ := ext.LookupScalarFunction(et.ScalarFunction.FunctionReference, c)
+			decl, _ = ext.LookupScalarFunction(et.ScalarFunction.FunctionReference, c)
+		}
```

Comment:
Can't this return an error if the scalar function isn't found? Should that be checked here and then returned on error?

Reply:
What happens down-the-road if decl is nil for ScalarFunction? In Java-land, we load the default extension collections statically up front, so they'd always be there - so in this case we'd error if we don't find it.

I think, in general, it's better to error in these cases early. Otherwise, strange things will happen down-the-road and it'll be difficult to trace it back to this.

Reply:
Make a lot of sense. I'll have it skip this check if the collection passed in is nil, but if the collection is non-nil it'll error if it can't find it.

I do plan on having the default extension collections eventually statically loaded, I just haven't implemented that yet.

Reply:
updated to check  if collection is nil and if it isn't, we error if we can't find the function